src/mono/wasm/debugger/BrowserDebugProxy/MonoProxy.cs

```diff
@@ -442,23 +442,20 @@ protected override async Task<bool> AcceptCommand(MessageId id, string method, J
                 case "Debugger.setPauseOnExceptions":
                     {
                         string state = args["state"].Value<string>();
-                        if (!context.IsRuntimeReady)
+                        switch (state)
                         {
-                            context.PauseOnCaught = false;
-                            context.PauseOnUncaught = false;
-                            switch (state)
-                            {
-                                case "all":
-                                    context.PauseOnCaught = true;
-                                    context.PauseOnUncaught = true;
-                                    break;
-                                case "uncaught":
-                                    context.PauseOnUncaught = true;
-                                    break;
-                            }
+                            case "all":
+                                context.PauseOnExceptions = PauseOnExceptionsKind.All;
+                                break;
+                            case "uncaught":
+                                context.PauseOnExceptions = PauseOnExceptionsKind.Uncaught;
+                                break;
+                            case "none":
+                                context.PauseOnExceptions = PauseOnExceptionsKind.None;
+                                break;
                         }
-                        else
-                            await SdbHelper.EnableExceptions(id, state, token);
+                        if (context.IsRuntimeReady)
+                            await SdbHelper.EnableExceptions(id, context.PauseOnExceptions, token);
```

Comment:
Is the reason for not using the same checks as line 1228 (`if (context.PauseOnExceptions != PauseOnExceptionsKind.None && context.PauseOnExceptions != PauseOnExceptionsKind.Unset)`):

- it shouldn't Unset, if we handle the default case in the switch above
- and if it's None, then we do want to send that to *set* the state to that?

But in line 1228's case:
- if it's None, then we need to send it because that's the default? nit: It might useful to send in case of None anyway, just to "enforce the default".

Reply:
In line 1228 we don't need to send PauseOnExceptionsKind.None because this is the default behavior on startup. 
If we receive from Debugger.setPauseOnExceptions and runtime is ready we maybe need to change the behavior to none if the behavior before wasn't none.

Reply:
I don't think we need to send none when the VM is started only to "enforce the default".